packages/main/src/RPA/Hubspot.py

```diff
@@ -257,9 +270,160 @@ def create_search_object(self, words: List[str] = None):
+        class does not validate the provided object type, that must be
+        done by the caller.
+        """
+        return self._object_type
+
+    @object_type.setter
+    def object_type(self, object_type: str):
+        self._object_type = object_type
+
+    @property
+    def inputs(self) -> Union[List[CreateObjectInput], List[UpdateObjectInput]]:
+        """A list of inputs to be sent into the Batch API, returned type
+        depends on ``mode``. Can be set directly, but the ``add_input`` and
+        ``extend_inputs`` methods are more useful.
+        """
+        return self._inputs
+
+    @inputs.setter
+    def inputs(self, inputs: Union[List[CreateObjectInput], List[UpdateObjectInput]]):
+        if isinstance(inputs[0], CreateObjectInput):
+            self.mode = BatchMode.CREATE
+        elif isinstance(inputs[0], UpdateObjectInput):
+            self.mode = BatchMode.UPDATE
+        else:
+            raise TypeError(
+                "Expected a list of type "
+                "SimplePublicObjectInput or SimplePublicObjectBatchInput,"
+                f"but received type {type(inputs[0])}"
+            )
+        self._inputs = inputs
+
+    @inputs.deleter
+    def inputs(self):
+        self._inputs = []
+
+    def _prevent_mode_change_with_inputs(self, other_mode: BatchMode):
+        if (
+            self.mode is not other_mode
+            and self.mode is not BatchMode.INVALID
+            and len(self.inputs) > 0
+        ):
+            raise ValueError(
+                f"This batch is in {str(self.mode)} mode and already has items, "
+                f"it is not possible to implicity switch modes to {str(other_mode)}"
+            )
+
+    def add_input(
+        self,
+        properties: Dict[str, str],
+        object_id: str = None,
+    ):
+        """Add the provided dictionary of Hubspot properties as an input
+        to this batch's ``inputs`` property. If ``object_id`` is provided, the
+        input will be considered an ``UPDATE`` mode input.
+        """
+        if object_id:
+            self._prevent_mode_change_with_inputs(BatchMode.UPDATE)
+            new_input = UpdateObjectInput(properties, object_id)
+            if self.mode is BatchMode.INVALID:
+                self.mode = BatchMode.UPDATE
+        else:
+            self._prevent_mode_change_with_inputs(BatchMode.CREATE)
+            new_input = CreateObjectInput(properties)
+            if self.mode is BatchMode.INVALID:
+                self.mode = BatchMode.CREATE
+        self._inputs.append(new_input)
+
+    def extend_inputs(
+        self, properties: List[Dict[str, str]], ids: List[str] = None
+    ) -> None:
+        """Extends this batch's ``inputs`` based on the provided list of
+        ``properties`` and ``ids``. If ``ids`` is provided, it will be
+        zipped with the provided ``properties``. ``UPDATE`` mode is assumed
+        in such a case.
+        """
+        if ids:
+            if len(ids) > 0:
```

Comment:
I feel this `if ids` already ensures the condition of having one or more elements, so why double check?

Reply:
Because it was a hold over when I refactored it, you are right.